mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala

```diff
@@ -129,6 +136,22 @@ final class DecisionTreeRegressionModel private[ml] (
     rootNode.predictImpl(features).prediction
   }
 
+  /** We need to update this function if we ever add other impurity measures. */
+  protected def predictVariance(features: Vector): Double = {
+    rootNode.predictImpl(features).impurityStats.calculate()
+  }
+
+  override protected def transformImpl(dataset: DataFrame): DataFrame = {
```

Comment:
Thanks for pointing out that predictionCol gets checked in transform.  Unfortunately, that means that turning off predictionCol will also turn off varianceCol.  This is annoying (and probably points to a flaw in the abstraction design), but you'll probably have to override both transform and transformImpl to get the desired behavior.